pkg/algorithm/cell_allocation.go

```diff
@@ -78,6 +79,70 @@ func buddyAlloc(
 	return false
 }
 
+// after buddyAlloc failed to allocate cells,
+// try to split a higher level cell to get current level cells.
```

Comment:
try to split a higher level cell to get current level cells while it won't break safety guarantee?

Reply:
used Fan's comment